java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -415,68 +391,66 @@ public ClassTreeImpl completeClassDeclaration(
+  private static ClassTreeImpl newClassBody(Kind kind, InternalSyntaxToken openBraceSyntaxToken, Optional<List<JavaTree>> members, InternalSyntaxToken closeBraceTokenSyntaxToken) {
+    List<JavaTree> children = Lists.newArrayList();
     ImmutableList.Builder<Tree> builder = ImmutableList.builder();
 
-    InternalSyntaxToken openBraceSyntaxToken = InternalSyntaxToken.create(openBraceTokenAstNode);
     children.add(openBraceSyntaxToken);
     if (members.isPresent()) {
-      for (AstNode member : members.get()) {
+      for (JavaTree member : members.get()) {
         children.add(member);
 
         if (member instanceof VariableDeclaratorListTreeImpl) {
           for (VariableTreeImpl variable : (VariableDeclaratorListTreeImpl) member) {
             builder.add(variable);
           }
-        } else if (member instanceof Tree) {
-          builder.add((Tree) member);
+        } else {
+          builder.add(member);
         }
       }
     }
-    InternalSyntaxToken closeBraceTokenSyntaxToken = InternalSyntaxToken.create(closeBraceTokenAstNode);
     children.add(closeBraceTokenSyntaxToken);
 
     return new ClassTreeImpl(kind, openBraceSyntaxToken, builder.build(), closeBraceTokenSyntaxToken, children);
   }
 
-  public ClassTreeImpl newClassBody(AstNode openBraceTokenAstNode, Optional<List<AstNode>> members, AstNode closeBraceTokenAstNode) {
-    return newClassBody(Kind.CLASS, openBraceTokenAstNode, members, closeBraceTokenAstNode);
+  public ClassTreeImpl newClassBody(InternalSyntaxToken openBraceToken, Optional<List<JavaTree>> members, InternalSyntaxToken closeBraceToken) {
+    return newClassBody(Kind.CLASS, openBraceToken, members, closeBraceToken);
   }
 
   public ClassTreeImpl newEnumDeclaration(
-    AstNode enumTokenAstNode,
-    AstNode identifierAstNode,
-    Optional<Tuple<AstNode, QualifiedIdentifierListTreeImpl>> implementsClause,
-    AstNode openBraceTokenAstNode,
+    JavaTree enumToken,
+    JavaTree identifierToken,
+    Optional<Tuple<JavaTree, QualifiedIdentifierListTreeImpl>> implementsClause,
+    InternalSyntaxToken openBraceToken,
     Optional<List<EnumConstantTreeImpl>> enumConstants,
-    Optional<AstNode> semicolonTokenAstNode,
-    Optional<List<AstNode>> enumDeclarations,
-    AstNode closeBraceTokenAstNode) {
+    Optional<InternalSyntaxToken> semicolonToken,
+    Optional<List<JavaTree>> enumDeclarations,
+    InternalSyntaxToken closeBraceToken) {
 
-    ImmutableList.Builder<AstNode> members = ImmutableList.builder();
+    ImmutableList.Builder<JavaTree> members = ImmutableList.builder();
     if (enumConstants.isPresent()) {
       for (EnumConstantTreeImpl enumConstant : enumConstants.get()) {
         members.add(enumConstant);
       }
     }
-    if (semicolonTokenAstNode.isPresent()) {
+    if (semicolonToken.isPresent()) {
       // TODO This is a hack
-      members.add(semicolonTokenAstNode.get());
+//      members.add(semicolonToken.get());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)